### PR TITLE
Draft: Rewrite process

### PR DESCRIPTION
**Pull request by Scriptkiddi**
Add more classes change flow structure to allow for easier adding of merge straegies

### CODE REVIEW

File: nixpkgs_merge_bot/database.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class Database:
+    def __init__(self, settings: Settings) -> None:
+        self.datbase_store_path = Path(settings.database_path)
+        if not self.datbase_store_path.exists():
+            self.datbase_store_path.mkdir()
+
+    def add(self, key: str, value: str) -> None:
+        path = self.datbase_store_path / key
+        if not path.exists():
+            path.mkdir()
```
Comment (Lassulus) [inline nixpkgs_merge_bot/database.py:15]:
```suggestion
        path.mkdir(parents=True, exists_ok=True)
```

File: nixpkgs_merge_bot/merging_startegies/merging_strategy.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+class MergingStrategyTemplate:
+    def __init__(self, client: GithubClient, settings: Settings) -> None:
+        self.github_client: GithubClient = client
+        self.settings: Settings = settings
+
+    def run_technical_limits_check(
+        self, pull_request: PullRequest
+    ) -> tuple[bool, list[str]]:
+        # Replace this to get rid of the dry-run mode
+        result = True
```
Comment (Lassulus) [inline nixpkgs_merge_bot/merging_startegies/merging_strategy.py:20]:
```suggestion
        result = False
```

**Review by Lassulus — APPROVED** [review]

**Comment by Lassulus** [comment]
@mergify queue

File: nixpkgs_merge_bot/commands/merge.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+from ..settings import Settings
+from ..webhook.http_response import HttpResponse
+from ..webhook.issue_comment import issue_response
+
+log = logging.getLogger(__name__)
+
+
+def process_pull_request_status(
+    client: GithubClient, pull_request: PullRequest
+) -> tuple[bool, bool, bool, list[str]]:
```
[the remark this answers is not in the document]
Comment (Scriptkiddi) [reply nixpkgs_merge_bot/commands/merge.py:17]: thanks will do it tomorrow

**Pull request merged**